website/client-old/js/controllers/inventoryCtrl.js

```diff
@@ -383,21 +382,19 @@ habitrpg.controller("InventoryCtrl",
     };
 
     $scope.shouldShowPremiumMountRow = function (potion) {
-      potion = Content.premiumHatchingPotions[potion].key;
       var pets = Object.keys(user.items.pets);
-      return pets.find(function (pet) {
-        return pet.indexOf(potion) !== -1;
+      return pets.find(function (petKey) {
```

Comment:
The difference is that the functions in the `$scope` can be used by anything else with access to that `$scope` (i.e., any views that use this controller), whereas the `var functionName` functions are helper functions that are not visible outside this file.